tests/app/main/views/test_agreements.py

```diff
@@ -52,35 +47,98 @@ def test_happy_path_g8(self, data_api_client):
+            a_elems[0].xpath("normalize-space(string())"),
+            mi_elems[0].xpath("normalize-space(string())"),
+        )
+
+    def test_happy_path_all_g8(self, data_api_client):
+        data_api_client.get_framework.return_value = self.load_example_listing('framework_response')
+        data_api_client.find_framework_suppliers.return_value = self.find_framework_suppliers_return_value_g8
+
         response = self.client.get('/admin/agreements/g-cloud-8')
         page = html.fromstring(response.get_data(as_text=True))
 
         assert response.status_code == 200
-        rows = page.cssselect('.search-result')
-
-        def unpack_search_result(elem):
-            a_elems = elem.cssselect(".search-result-title a")
-            assert len(a_elems) == 1
-            mi_elems = elem.cssselect(".search-result-metadata-item")
-            assert len(mi_elems) == 1
-            return (
-                urlunparse(("", "",) + urlparse(a_elems[0].attrib["href"])[2:]),
-                a_elems[0].text.strip(),
-                mi_elems[0].text.strip(),
-            )
-        assert tuple(unpack_search_result(result) for result in page.cssselect('.search-result')) == (
+
+        call_args = data_api_client.find_framework_suppliers.call_args
+        assert call_args[0] == ("g-cloud-8",)
+        # slightly elaborate assertion here to allow for missing kwargs defaulting to None
+        assert all(call_args[1].get(key) == value for key, value in (
+            ("agreement_returned", True),
+            ("statuses", None),
+        ))
+
+        assert tuple(self._unpack_search_result(result) for result in page.cssselect('.search-result')) == (
+            (
+                "/admin/suppliers/11112/agreements/g-cloud-8",
+                "My other supplier",
+                "Submitted: Friday 30 October 2015 at 01:01",
+            ),
+            (
+                "/admin/suppliers/11111/agreements/g-cloud-8",
+                "My Supplier",
+                "Submitted: Sunday 1 November 2015 at 01:01",
+            ),
+        )
+
+        assert page.xpath("//*[@class='status-filters']//li[normalize-space(string())='All']")
+
+        assert tuple(
+            (parse_qs(urlparse(a_element.attrib["href"]).query), a_element.xpath("normalize-space(string())"))
+            for a_element in page.cssselect('.status-filters a')
+        ) == tuple(
+            ({"status": [status_key]}, status_label)
+            for status_key, status_label in iteritems(status_labels)
+        )
+
+    def test_happy_path_notall_g8(self, data_api_client):
+        data_api_client.get_framework.return_value = self.load_example_listing('framework_response')
+        data_api_client.find_framework_suppliers.return_value = self.find_framework_suppliers_return_value_g8
+
+        # choose the second status (if there is one, otherwise first)
+        chosen_status_key = tuple(iterkeys(status_labels))[:2][-1]
```

Comment:
Yeah I'm just trying to make it so that someone can come along and drop another status option in there or remove one and not have to come along and fix the tests.